### PR TITLE
Stories to XML

### PR DESCRIPTION
**Pull request by clayboone**
Convert all the `message.txt` and `meta.json` into action XML files. This is in preparation for the new stories to come with more attributes. At the cost of a tiny amount of complexity for the writers, a great deal of complexity for developers has been removed. The XML is all validated and tests have been written to prove them.

This doesn't add any functionality that wasn't there already, like "tiers" or "missed actions", those are coming next.

I've updated some requirements and added a new one, so make sure to `pip install` again.

### CODE REVIEW

File: pombot/data/pom_wars/actions/__init__.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+                etree.parse(str(path)).getroot()  # pylint: disable=c-extension-no-member
+                for path in Locations.POMWARS_ACTIONS_DIR.rglob("*.xml"))))
+
+    @abstractproperty
+    def TAGS(self) -> List[str]:  # pylint: disable=invalid-name
+        """Tags which represent actions of the child's type in the XML."""
+
+    @abstractproperty
+    def TYPE(self) -> Callable:  # pylint: disable=invalid-name
+        """Tags which represent actions of the child's type in the XML."""
```
Comment (clayboone) [inline pombot/data/pom_wars/actions/__init__.py:30]: eh

File: pombot/data/pom_wars/actions/__init__.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+class _Attacks(_XMLLoader):
+    TAGS = ["normal_attack", "heavy_attack"]
+    TYPE = Attack
+
+    def get_random(self, *, critical: bool, heavy: bool) -> Attack:  # pylint: disable=arguments-differ
+        """Return an Attack at random from the XMLs."""
+        elements = (e for e in self._xml_elements
+                    if e.tag == "heavy_attack") if heavy else (
+                        e for e in self._xml_elements
+                        if e.tag == "normal_attack")
```
Comment (clayboone) [inline pombot/data/pom_wars/actions/__init__.py:46]: hmm

File: pombot/lib/pom_wars/team.py [hunk 3]
```diff
@@ -62,3 +50,21 @@ async def attack_count(self) -> int:
     async def population(self) -> int:
         """The team's population."""
         return await Storage.count_rows_in_table(Config.USERS_TABLE, team=self.value)
+
+
+def get_user_team(user: User) -> Team:
+    """Find a Discord user's tead based on their roles.
```
Comment (clayboone) [inline pombot/lib/pom_wars/team.py:56]: heh

File: pombot/lib/pom_wars/types.py [hunk 4]
```diff
@@ -1,137 +1,97 @@
...
+        self._is_critical = is_critical
 
     @property
     def damage(self):
-        """The configured base damage for this action."""
-        base_damage = Pomwars.BASE_DAMAGE_FOR_NORMAL_ATTACKS
-
-        if self.is_heavy:
-            base_damage = Pomwars.BASE_DAMAGE_FOR_HEAVY_ATTACKS
+        """Return the total damange this attack produces after heavy- and
```
Comment (clayboone) [inline pombot/lib/pom_wars/types.py:21]: ugh!

File: requirements.txt [hunk 5]
```diff
@@ -1,6 +1,7 @@
 # Prod
 aiomysql>=0.0.21
 discord.py>=1.7.1
+lxml>=4.6.3
```
Comment (clayboone) [inline requirements.txt:4]: oh

File: pombot/data/actions-to-xml.py [hunk 6]
```diff
@@ -1,38 +0,0 @@
-from pathlib import Path
```
Comment (clayboone) [inline pombot/data/actions-to-xml.py]: For the curious, I wrote a hacky script to convert all the `action_dir/[message.txt|meta.json]` into a single XML here.

File: tests/test_data_pomwars_actions.py [hunk 7]
```diff
@@ -0,0 +1,210 @@
...
+                </team>
+                <team name="Viking">
+                    <normal_attack is_critical="true">{stories["vk_crit_norm"]}</normal_attack>
+                    <heavy_attack is_critical="true">{stories["vk_crit_heav"]}</heavy_attack>
+                    <normal_attack>{stories["vk_norm"]}</normal_attack>
+                    <heavy_attack>{stories["vk_heav"]}</heavy_attack>
+                    <defend>{stories["vk_def"]}</defend>
+                </team>
+                <bribe>{stories["br"]}</bribe>
+            </actions>
```
Comment (clayboone) [inline tests/test_data_pomwars_actions.py:184]: This.

**Review by debashisbiswas — APPROVED** [review]
Great! Looks much cleaner without those unused JSON files and their related code too.

**Pull request merged**